packages/grid/_modules_/grid/hooks/features/columns/useGridColumns.ts

```diff
@@ -58,10 +58,24 @@ function updateColumnsWidth(columns: GridColumns, viewportWidth: number) {
       if (!column.flex) {
         return column;
       }
+
+      const minColumnWidth = column.minWidth! ?? GRID_STRING_COL_DEF.width;
+      const flexColumnWidth = Math.floor(flexMultiplier * column.flex!);
+      let newWidth = 0;
+
+      if (viewportWidth > 0) {
+        if (column.minWidth && column.minWidth > flexColumnWidth) {
+          newWidth = column.minWidth;
+        } else {
+          newWidth = flexColumnWidth;
+        }
+      } else {
+        newWidth = minColumnWidth;
+      }
```

Comment:
A small simplification:

```suggestion
      let newWidth = column.minWidth! ?? GRID_STRING_COL_DEF.width;
      if (viewportWidth > 0) {
        const flexColumnWidth = Math.floor(flexMultiplier * column.flex!);
        newWidth =
          column.minWidth && column.minWidth > flexColumnWidth ? column.minWidth : flexColumnWidth;
      }
```